blueocean-rest-impl/src/test/java/io/jenkins/blueocean/util/HttpRequest.java

```diff
@@ -0,0 +1,225 @@
+    private String requestUrl;
+    private String contentType;
+    private String requestBody;
+    private String username;
+    private String password;
+
+
+    private HttpRequest() {}
+
+    public static HttpRequest build(String baseUrl) {
+        Preconditions.checkState(StringUtils.isNotBlank(baseUrl), "baseUrl is required");
+        HttpRequest builder = new HttpRequest();
+        builder.baseUrl(baseUrl);
+        return builder;
+    }
+
+    public HttpRequest baseUrl(@Nonnull String baseUrl) {
+        this.baseUrl = baseUrl;
+        return this;
+    }
+
+    public HttpRequest GET(String url) {
+        this.method = GET;
+        this.requestUrl = url;
+        return this;
+    }
+
+    public HttpRequest POST(String url) {
+        this.method = Method.POST;
+        this.requestUrl = url;
+        return this;
+    }
+
+    public HttpRequest PUT(String url) {
+        this.method = Method.PUT;
+        this.requestUrl = url;
+        return this;
+    }
+
+    public HttpRequest PATCH(String url) {
+        this.method = Method.PATCH;
+        this.requestUrl = url;
+        return this;
+    }
+
+    public HttpRequest DELETE(String url) {
+        this.method = Method.DELETE;
+        this.requestUrl = url;
+        return this;
+    }
+
+    public HttpRequest urlPart(String key, String value) {
+        urlParts.put(key, value);
+        return this;
+    }
+
+    public HttpRequest header(String name, String value) {
+        headers.put(name, value);
+        return this;
+    }
+
+    public HttpRequest auth(String username, String password) {
+        this.username = username;
+        this.password = password;
+        return this;
+    }
+
+    public HttpRequest contentType(String contentType) {
+        this.contentType = contentType;
+        return this;
+    }
+
+    public HttpRequest bodyJson(Object object) {
+        if (object != null) {
+            contentType(JSON);
+            requestBody = JsonConverter.toJson(object);
+        } else {
+            contentType(null);
+            requestBody = null;
+        }
+        return this;
+    }
+
+    public HttpRequest status(int statusCode) {
+        this.expectedStatus = statusCode;
+        return this;
+    }
+
+    public <T> T asObject(Class<T> clazz) throws IOException {
+        return JsonConverter.toJava(asInputStream(), clazz);
+    }
+
+    public String asText() throws IOException {
+        return executeInternal().asString();
+    }
+
+    public String asText(String charset) throws IOException {
+        return executeInternal().asString(Charset.forName(charset));
+    }
+
+    public InputStream asInputStream() throws IOException {
+        return executeInternal().asStream();
+    }
+
+    public byte[] asBytes() throws IOException {
+        return executeInternal().asBytes();
+    }
+
+    public void execute() throws IOException {
+        executeInternal();
+    }
+
+    public Content executeInternal() throws IOException {
```

Comment:
private?

Reply:
Good catch, silly mistake, fixed in 360cfd8